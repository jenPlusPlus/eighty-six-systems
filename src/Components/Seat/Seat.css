.seat-order-container {
  height: 93%;
  width: 95%;
  overflow-y: scroll;
  margin-top: 5px;
  background-color: #EFEFF0;
  color: #33202A;
  border-radius: 5px;
  margin: auto;
  padding: 5px 0 3px 0; }

.menu-item {
  color: #33202A;
  margin-bottom: 5px;
  text-align: left; }

.edit-order-item-button {
  display: none; }

.remove-order-item-button {
  background-color: #DADADB;
  color: #A63D40;
  float: left;
  border-radius: 50%;
  border: 1px solid #C68385;
  width: 17px;
  height: 17px;
  font-size: 15px;
  margin: 0 5px 0 3px; }
