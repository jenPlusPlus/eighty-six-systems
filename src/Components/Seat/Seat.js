import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Seat extends Component {
  constructor() {
    super();
  }

  handleClick() {
    this.props.addCurrentSeat({
      tableNumber: this.props.currentTableNumber,
      seatNumber: this.props.seat.seatNumber,
      order: []
    });
  }

  render() {
    return (
      <div className='seat'
        onClick={() => this.handleClick()}>
        <h3>{this.props.seat.seatNumber}</h3>
      </div>
    );
  }
}

Seat.propTypes = {
  currentTableNumber: PropTypes.string,
  seat: PropTypes.object,
  addCurrentSeat: PropTypes.func
};

export default Seat;
