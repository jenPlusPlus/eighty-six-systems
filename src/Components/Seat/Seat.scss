.orders-container {
  height: 93%;
  width: 95%;
  overflow-y: scroll;
  margin-top: 5px;
  background-color: #EFEFF0;
  color: #33202A;
  border-radius: 5px;
  margin:auto;
  padding: 5px 0 3px 0;
}

.menu-item {
  color: #33202A;
  margin-bottom: 5px;
}

.prev-ordered-menu-item {
  color: #5F5F60;
  margin-bottom: 5px;
}

.prev-orders-list {
  color: #5F5F60;
  margin: 5px 0 5px 0;
  padding-top: 5px;
}

.has-current-order {
  border-top: dashed 1px #5F5F60;
}

.edit-order-item-button {
  display: none;
}

.remove-order-item-button {
  background-color: #DADADB;
  color: #A63D40;
  // float: left;
  border-radius: 50%;
  border: 1px solid #C68385;
  width: 17px;
  height: 17px;
  font-size: 15px;
  margin: 0 5px 0 3px;
}

.has-no-previous-orders {
  display: none;
}
