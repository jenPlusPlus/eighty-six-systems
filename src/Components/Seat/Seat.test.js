import { shallow } from 'enzyme';
import Seat from './Seat';
import ReactDOM from 'react-dom';
import React from 'react';

describe('Seat snapshot', () => {

  const initialState = {
    currentUser: {name: 'Jen', loginCode: '0000'},
    currentTable: {tableNumber: '22'},
    tables: [{
      tableNumber: '22',
      seats: [{
        seatNumber: '1',
        order: [
          {item: 'Lamb', price: 29},
          {item: 'Burger', price: 19},
          {item:'Mahi Mahi', price: 27}
        ]
      }]
    }],
    currentSeatOrder: [
      {item: 'Lamb', price: 29},
      {item: 'Burger', price: 19},
      {item:'Mahi Mahi', price: 27}
    ],
    currentTableOrder: [
      {
        seatNumber: '1',
        currentSeatOrder: [
          {item: 'Lamb', price: 29},
          {item: 'Burger', price: 19},
          {item:'Mahi Mahi', price: 27}
        ]
      },
      {
        seatNumber: '2',
        currentSeatOrder: [
          {item: 'Scallops', price: 30}
        ]
      }
    ]
  };

  Date.now = jest.fn(() => 1482363367071);

  it('should always match the snapshot', () => {
    const wrapper = shallow(<Seat
      currentUser={initialState.currentUser}
      currentTable={initialState.currentTable}
      tables={initialState.tables}
      currentSeatOrder={initialState.currentSeatOrder}
      currentTableOrder={initialState.currentTableOrder}
      key={1482363367071}
      seat={{
        tableNumber: '22',
        seatNumber: '1',
        order: [
        {item: 'Lamb', price: 29},
        {item: 'Burger', price: 19},
        {item:'Mahi Mahi', price: 27}
        ]
      }}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
