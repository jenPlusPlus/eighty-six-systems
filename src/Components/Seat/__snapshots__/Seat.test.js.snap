// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seat snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Seat
    currentSeatOrder={
        Array [
            Object {
              "item": "Lamb",
              "price": 29,
            },
            Object {
              "item": "Burger",
              "price": 19,
            },
            Object {
              "item": "Mahi Mahi",
              "price": 27,
            },
          ]
    }
    currentTable={
        Object {
            "tableNumber": "22",
          }
    }
    currentTableOrder={
        Array [
            Object {
              "currentSeatOrder": Array [
                Object {
                  "item": "Lamb",
                  "price": 29,
                },
                Object {
                  "item": "Burger",
                  "price": 19,
                },
                Object {
                  "item": "Mahi Mahi",
                  "price": 27,
                },
              ],
              "seatNumber": "1",
            },
            Object {
              "currentSeatOrder": Array [
                Object {
                  "item": "Scallops",
                  "price": 30,
                },
              ],
              "seatNumber": "2",
            },
          ]
    }
    currentUser={
        Object {
            "loginCode": "0000",
            "name": "Jen",
          }
    }
    seat={
        Object {
            "order": Array [
              Object {
                "item": "Lamb",
                "price": 29,
              },
              Object {
                "item": "Burger",
                "price": 19,
              },
              Object {
                "item": "Mahi Mahi",
                "price": 27,
              },
            ],
            "seatNumber": "1",
            "tableNumber": "22",
          }
    }
    tables={
        Array [
            Object {
              "seats": Array [
                Object {
                  "order": Array [
                    Object {
                      "item": "Lamb",
                      "price": 29,
                    },
                    Object {
                      "item": "Burger",
                      "price": 19,
                    },
                    Object {
                      "item": "Mahi Mahi",
                      "price": 27,
                    },
                  ],
                  "seatNumber": "1",
                },
              ],
              "tableNumber": "22",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/0000/tables/22/1/menu"
>
          <h3>
                    1
          </h3>
</Link>,
        <ul
          className="seat-order-container"
>
          <li
                    className="menu-item"
          >
                    Lamb
                    <button
                              className="edit-order-item-button"
                    >
                              Edit
                    </button>
                    <button
                              className="remove-order-item-button"
                              onClick={[Function]}
                    >
                              X
                    </button>
          </li>
          <li
                    className="menu-item"
          >
                    Burger
                    <button
                              className="edit-order-item-button"
                    >
                              Edit
                    </button>
                    <button
                              className="remove-order-item-button"
                              onClick={[Function]}
                    >
                              X
                    </button>
          </li>
          <li
                    className="menu-item"
          >
                    Mahi Mahi
                    <button
                              className="edit-order-item-button"
                    >
                              Edit
                    </button>
                    <button
                              className="remove-order-item-button"
                              onClick={[Function]}
                    >
                              X
                    </button>
          </li>
</ul>,
      ],
      "className": "seat",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h3>
            1
</h3>,
          "replace": false,
          "to": "/0000/tables/22/1/menu",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1",
          },
          "ref": null,
          "rendered": "1",
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="menu-item"
>
              Lamb
              <button
                            className="edit-order-item-button"
              >
                            Edit
              </button>
              <button
                            className="remove-order-item-button"
                            onClick={[Function]}
              >
                            X
              </button>
</li>,
            <li
              className="menu-item"
>
              Burger
              <button
                            className="edit-order-item-button"
              >
                            Edit
              </button>
              <button
                            className="remove-order-item-button"
                            onClick={[Function]}
              >
                            X
              </button>
</li>,
            <li
              className="menu-item"
>
              Mahi Mahi
              <button
                            className="edit-order-item-button"
              >
                            Edit
              </button>
              <button
                            className="remove-order-item-button"
                            onClick={[Function]}
              >
                            X
              </button>
</li>,
          ],
          "className": "seat-order-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1482363367071",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Lamb",
                <button
                  className="edit-order-item-button"
>
                  Edit
</button>,
                <button
                  className="remove-order-item-button"
                  onClick={[Function]}
>
                  X
</button>,
              ],
              "className": "menu-item",
            },
            "ref": null,
            "rendered": Array [
              "Lamb",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "edit-order-item-button",
                },
                "ref": null,
                "rendered": "Edit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "className": "remove-order-item-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1482363367072",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Burger",
                <button
                  className="edit-order-item-button"
>
                  Edit
</button>,
                <button
                  className="remove-order-item-button"
                  onClick={[Function]}
>
                  X
</button>,
              ],
              "className": "menu-item",
            },
            "ref": null,
            "rendered": Array [
              "Burger",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "edit-order-item-button",
                },
                "ref": null,
                "rendered": "Edit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "className": "remove-order-item-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1482363367073",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Mahi Mahi",
                <button
                  className="edit-order-item-button"
>
                  Edit
</button>,
                <button
                  className="remove-order-item-button"
                  onClick={[Function]}
>
                  X
</button>,
              ],
              "className": "menu-item",
            },
            "ref": null,
            "rendered": Array [
              "Mahi Mahi",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "edit-order-item-button",
                },
                "ref": null,
                "rendered": "Edit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "className": "remove-order-item-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/0000/tables/22/1/menu"
>
            <h3>
                        1
            </h3>
</Link>,
          <ul
            className="seat-order-container"
>
            <li
                        className="menu-item"
            >
                        Lamb
                        <button
                                    className="edit-order-item-button"
                        >
                                    Edit
                        </button>
                        <button
                                    className="remove-order-item-button"
                                    onClick={[Function]}
                        >
                                    X
                        </button>
            </li>
            <li
                        className="menu-item"
            >
                        Burger
                        <button
                                    className="edit-order-item-button"
                        >
                                    Edit
                        </button>
                        <button
                                    className="remove-order-item-button"
                                    onClick={[Function]}
                        >
                                    X
                        </button>
            </li>
            <li
                        className="menu-item"
            >
                        Mahi Mahi
                        <button
                                    className="edit-order-item-button"
                        >
                                    Edit
                        </button>
                        <button
                                    className="remove-order-item-button"
                                    onClick={[Function]}
                        >
                                    X
                        </button>
            </li>
</ul>,
        ],
        "className": "seat",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              1
</h3>,
            "replace": false,
            "to": "/0000/tables/22/1/menu",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
            },
            "ref": null,
            "rendered": "1",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="menu-item"
>
                Lamb
                <button
                                className="edit-order-item-button"
                >
                                Edit
                </button>
                <button
                                className="remove-order-item-button"
                                onClick={[Function]}
                >
                                X
                </button>
</li>,
              <li
                className="menu-item"
>
                Burger
                <button
                                className="edit-order-item-button"
                >
                                Edit
                </button>
                <button
                                className="remove-order-item-button"
                                onClick={[Function]}
                >
                                X
                </button>
</li>,
              <li
                className="menu-item"
>
                Mahi Mahi
                <button
                                className="edit-order-item-button"
                >
                                Edit
                </button>
                <button
                                className="remove-order-item-button"
                                onClick={[Function]}
                >
                                X
                </button>
</li>,
            ],
            "className": "seat-order-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1482363367071",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Lamb",
                  <button
                    className="edit-order-item-button"
>
                    Edit
</button>,
                  <button
                    className="remove-order-item-button"
                    onClick={[Function]}
>
                    X
</button>,
                ],
                "className": "menu-item",
              },
              "ref": null,
              "rendered": Array [
                "Lamb",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit",
                    "className": "edit-order-item-button",
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "X",
                    "className": "remove-order-item-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "X",
                  "type": "button",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1482363367072",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Burger",
                  <button
                    className="edit-order-item-button"
>
                    Edit
</button>,
                  <button
                    className="remove-order-item-button"
                    onClick={[Function]}
>
                    X
</button>,
                ],
                "className": "menu-item",
              },
              "ref": null,
              "rendered": Array [
                "Burger",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit",
                    "className": "edit-order-item-button",
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "X",
                    "className": "remove-order-item-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "X",
                  "type": "button",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1482363367073",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Mahi Mahi",
                  <button
                    className="edit-order-item-button"
>
                    Edit
</button>,
                  <button
                    className="remove-order-item-button"
                    onClick={[Function]}
>
                    X
</button>,
                ],
                "className": "menu-item",
              },
              "ref": null,
              "rendered": Array [
                "Mahi Mahi",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit",
                    "className": "edit-order-item-button",
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "X",
                    "className": "remove-order-item-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "X",
                  "type": "button",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
